shared-bindings/_pixelbuf/PixelBuf.c

```diff
@@ -87,23 +83,61 @@ STATIC mp_obj_t pixelbuf_pixelbuf_make_new(const mp_obj_type_t *type, size_t n_a
         { MP_QSTR_brightness, MP_ARG_OBJ, { .u_obj = mp_const_none } },
         { MP_QSTR_rawbuf, MP_ARG_OBJ, { .u_obj = mp_const_none } },
         { MP_QSTR_offset, MP_ARG_INT, { .u_int = 0 } },
-        { MP_QSTR_dotstar, MP_ARG_BOOL, { .u_bool = false } },
         { MP_QSTR_auto_write, MP_ARG_BOOL, {.u_bool = false} },
         { MP_QSTR_write_function, MP_ARG_OBJ, {.u_obj = mp_const_none} },
         { MP_QSTR_write_args, MP_ARG_OBJ, {.u_obj = mp_const_none} },
     };
     mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
     mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+    const char *byteorder = NULL;
+    pixelbuf_byteorder_details_t byteorder_details;
+    size_t bo_len;
+
+    if (!MP_OBJ_IS_STR(args[ARG_byteorder].u_obj))
+        mp_raise_TypeError(translate("byteorder is not a string"));
+
+    if (args[ARG_byteorder].u_obj == NULL)
+        byteorder = "BGR";
+    else 
+        byteorder = mp_obj_str_get_data(args[ARG_byteorder].u_obj, &bo_len);
+
+    if (bo_len < 3 || bo_len > 4)
+        mp_raise_ValueError(translate("Invalid byteorder string"));
+    strncpy(byteorder_details.order, byteorder, sizeof(byteorder_details.order));
```

Comment:
Why not hang onto the original object? Strings are immutable so it won't change.